pandas/io/tests/parser/test_network.py

```diff
@@ -7,31 +7,44 @@
 
 import os
 import nose
+import functools
 
 import pandas.util.testing as tm
 from pandas import DataFrame
 from pandas import compat
 from pandas.io.parsers import read_csv, read_table
 
 
-class TestUrlGz(tm.TestCase):
-
-    def setUp(self):
-        dirpath = tm.get_data_path()
-        localtable = os.path.join(dirpath, 'salaries.csv')
-        self.local_table = read_table(localtable)
-
-    @tm.network
-    def test_url_gz(self):
-        url = ('https://raw.github.com/pandas-dev/pandas/'
-               'master/pandas/io/tests/parser/data/salaries.csv.gz')
-        url_table = read_table(url, compression="gzip", engine="python")
-        tm.assert_frame_equal(url_table, self.local_table)
-
-    @tm.network
-    def test_url_gz_infer(self):
-        url = 'https://s3.amazonaws.com/pandas-test/salary.table.gz'
-        url_table = read_table(url, compression="infer", engine="python")
+class TestCompressedUrl(object):
+    
+    compression_to_extension = {
+        'gzip': '.gz',
+        'bz2': '.bz2',
+        'zip': '.zip',
+        'xz': '.xz',
```

Comment:
Did you want a specific test for invalid compression specified for a URL? As of the latest commit (bff0f3e2cdabfe02cfcbe95667a1a2ea1481fe1e), compression inference has been consolidated and invalid compression [raises an error](https://github.com/dhimmel/pandas/blob/bff0f3e2cdabfe02cfcbe95667a1a2ea1481fe1e/pandas/io/parsers.py#L374-L375) before compression is attempted.